include/circt/Support/LLVM.h

```diff
@@ -167,97 +169,102 @@ struct OpRewritePattern;
 
 using DefaultTypeStorage = TypeStorage;
 using OpAsmSetValueNameFn = function_ref<void(Value, StringRef)>;
-using NamedAttribute = std::pair<Identifier, Attribute>;
 
 namespace OpTrait {}
 
 } // namespace mlir
 
 // Import things we want into our namespace.
 namespace circt {
-using mlir::ArrayAttr;
-using mlir::Attribute;
-using mlir::Block;
-using mlir::BlockAndValueMapping;
-using mlir::BlockArgument;
-using mlir::BoolAttr;
-using mlir::Builder;
-using mlir::CallInterfaceCallable;
-using mlir::ConversionPattern;
-using mlir::ConversionPatternRewriter;
-using mlir::ConversionTarget;
-using mlir::DefaultTypeStorage;
-using mlir::DenseElementsAttr;
-using mlir::Diagnostic;
-using mlir::Dialect;
-using mlir::DialectAsmParser;
-using mlir::DialectAsmPrinter;
-using mlir::DictionaryAttr;
-using mlir::ElementsAttr;
-using mlir::failed;
-using mlir::failure;
-using mlir::FileLineColLoc;
-using mlir::FlatSymbolRefAttr;
-using mlir::FloatAttr;
-using mlir::FunctionType;
-using mlir::FusedLoc;
-using mlir::Identifier;
-using mlir::ImplicitLocOpBuilder;
-using mlir::IndexType;
-using mlir::InFlightDiagnostic;
-using mlir::IntegerAttr;
-using mlir::IntegerType;
-using mlir::Location;
-using mlir::LogicalResult;
-using mlir::MemRefAccess;
-using mlir::MemRefType;
-using mlir::MLIRContext;
-using mlir::ModuleOp;
-using mlir::MutableOperandRange;
-using mlir::NamedAttribute;
-using mlir::NamedAttrList;
-using mlir::NoneType;
-using mlir::OpAsmDialectInterface;
-using mlir::OpAsmParser;
-using mlir::OpAsmPrinter;
-using mlir::OpAsmSetValueNameFn;
-using mlir::OpBuilder;
-using mlir::OpConversionPattern;
-using mlir::OperandRange;
-using mlir::Operation;
-using mlir::OperationPass;
-using mlir::OperationState;
-using mlir::OpFoldResult;
-using mlir::OpOperand;
-using mlir::OpResult;
-using mlir::OpRewritePattern;
-using mlir::OwningModuleRef;
-using mlir::ParseResult;
-using mlir::Pass;
-using mlir::PatternRewriter;
-using mlir::Region;
-using mlir::RegionKind;
-using mlir::RewritePatternSet;
-using mlir::ShapedType;
-using mlir::SplatElementsAttr;
-using mlir::StringAttr;
-using mlir::succeeded;
-using mlir::success;
-using mlir::SymbolRefAttr;
-using mlir::SymbolTable;
-using mlir::SymbolTableCollection;
-using mlir::TupleType;
-using mlir::Type;
-using mlir::TypeAttr;
-using mlir::TypeConverter;
-using mlir::TypeID;
-using mlir::TypeRange;
-using mlir::TypeStorage;
-using mlir::UnknownLoc;
-using mlir::Value;
-using mlir::ValueRange;
-using mlir::VectorType;
-using mlir::WalkResult;
+// clang-tidy removes following using directives incorrectly. So force
+// clang-tidy to ignore them.
+// TODO: It is better to use `NOLINTBEGIN/END` comments to disable clang-tidy
+// than adding `NOLINT` to every line. `NOLINTBEGIN/END` will supported from
+// clang-tidy-14.
+using mlir::ArrayAttr;                 // NOLINT(misc-unused-using-decls)
```

Comment:
This is unfortunately ugly, but I don't know how else to handle it and us fighting with clang-tidy on code that _clearly compiles and is fine_ shouldn't hold up the PR.  We can try to figure out how to remove this in the future.

Reply:
Yes, agreed. It's definitely a bug of `misc-unused-using-decls`.